lib/Core/Executor.cpp

```diff
@@ -4697,8 +4703,45 @@ void Executor::run(ExecutionState *initialState) {
 
   objectManager->initialUpdate();
 
+  auto lastExecutionTime = std::chrono::steady_clock::now();
+  Delta dt;
+  std::thread consumer;
+  std::thread producer;
+  StatisticQueue StatQ;
+
+  dt.initPrevDelta(objectManager->StatisticMap);
+  ServerConnection sc;
+  sc.url = "http://localhost:8080/metrics";
+  sc.getUID();
+  std::atomic<bool> shouldStop = false;
+  if (sc.UID != "ServerNotValid") {
+    consumer = std::thread([&StatQ, &dt, &sc, &shouldStop]() {
+      while (!shouldStop) {
+        if (!StatQ.empty()) {
+          auto data = StatQ.pop();
+          sc.PostRequest(dt.SerializeDelMap(data, sc.UID));
+        }
+      }
+    });
+
+    producer =
+        std::thread([&lastExecutionTime, &dt, &StatQ, &shouldStop, this]() {
+          while (!shouldStop) {
+            auto currentTime = std::chrono::steady_clock::now();
+            if (std::chrono::duration_cast<std::chrono::milliseconds>(
+                    currentTime - lastExecutionTime)
+                    .count() >= 100) {
```

Comment:
Additionally, the approach of performing checks in a spinlock manner does not seem to be a good idea (because the CPU core will most likely just keep spinning until 100ms has passed). Consider using timers.